model/reliability/query_test.go

```diff
@@ -268,3 +281,376 @@ func TestValidateForTaskReliability(t *testing.T) {
 	assert.Contains(message, "Invalid Limit value")
 	assert.Contains(message, "Invalid Significance value")
 }
+
+func TestGetTaskReliabilityGrouped(t *testing.T) {
+	require := require.New(t)
+	filter := createValidFilter()
+	task1 := "task1"
+	task2 := "task2"
+
+	variant1 := "v1"
+	variant2 := "v2"
+
+	distro1 := "d1"
+	distro2 := "d2"
+
+	err := clearCollection()
+	require.NoError(err)
+
+	require.NoError(insertDailyTaskStats(project, "r1", task1, variant1, distro1, day1, 10, 5, 1, 1, 1, 2, 10.5))
+	require.NoError(insertDailyTaskStats(project, "r1", task1, variant1, distro1, day2.Add(-1*time.Hour), 20, 7, 7, 0, 0, 0, 20.0))
+
+	require.NoError(insertDailyTaskStats(project, "r1", task2, variant1, distro2, day1, 10, 5, 1, 1, 1, 2, 10.5))
+	require.NoError(insertDailyTaskStats(project, "r1", task2, variant1, distro2, day2.Add(-1*time.Hour), 20, 7, 7, 0, 0, 0, 20.0))
+
+	docs, err := GetTaskReliabilityScores(filter)
+	require.NoError(err)
+	require.NotEqual(len(docs), 0)
+
+	filter.StatsFilter.Tasks = []string{task1}
+	filter.StatsFilter.BuildVariants = []string{variant1, variant2}
+	filter.StatsFilter.Distros = []string{distro1, distro2}
+	docs, err = GetTaskReliabilityScores(filter)
+	require.NoError(err)
+	require.Len(docs, 2)
+	require.NotEqual(docs[0].Date, day1)
+	require.NotEqual(docs[1].Date, day2)
```

Comment:
I'd isolate various assertions into seperate subtests so it's more clear if/when things fail what we're testing. 

also I'd either delete or uncomment the lines that follow.